plugins/network-elements/tungsten/src/main/java/org/apache/cloudstack/network/tungsten/api/command/ConfigTungstenServiceCmd.java

```diff
@@ -0,0 +1,169 @@
+
+                    Map<Network.Service, Network.Provider> tungstenServiceProvider = new HashMap<>();
+                    tungstenServiceProvider.put(Network.Service.Dhcp, Network.Provider.Tungsten);
+                    tungstenServiceProvider.put(Network.Service.Dns, Network.Provider.Tungsten);
+                    tungstenServiceProvider.put(Network.Service.UserData, Network.Provider.Tungsten);
+                    tungstenServiceProvider.put(Network.Service.SourceNat, Network.Provider.Tungsten);
+                    tungstenServiceProvider.put(Network.Service.StaticNat, Network.Provider.Tungsten);
+                    tungstenServiceProvider.put(Network.Service.Connectivity, Network.Provider.Tungsten);
+                    tungstenServiceProvider.put(Network.Service.Firewall, Network.Provider.Tungsten);
+                    tungstenServiceProvider.put(Network.Service.Lb, Network.Provider.Tungsten);
+                    tungstenServiceProvider.put(Network.Service.PortForwarding, Network.Provider.Tungsten);
+
+                    for (Network.Service service : tungstenServiceProvider.keySet()) {
+                        NetworkOfferingServiceMapVO networkOfferingServiceMapVO = new NetworkOfferingServiceMapVO(
+                            networkOfferingVO.getId(), service, tungstenServiceProvider.get(service));
+                        _networkOfferingServiceMapDao.persist(networkOfferingServiceMapVO);
+                    }
+                }
+
+                PhysicalNetworkServiceProviderVO physicalNetworkServiceProvider =
+                    _physicalNetworkServiceProviderDao.findByServiceProvider(
+                    physicalNetworkId, Network.Provider.Tungsten.getName());
+                physicalNetworkServiceProvider.setState(PhysicalNetworkServiceProvider.State.Enabled);
+                _physicalNetworkServiceProviderDao.persist(physicalNetworkServiceProvider);
+
+                Network publicNetwork = _networkModel.getSystemNetworkByZoneAndTrafficType(zoneId,
+                    Networks.TrafficType.Public);
+                NetworkServiceMapVO publicNetworkServiceMapVO = new NetworkServiceMapVO(publicNetwork.getId(),
+                    Network.Service.Connectivity, Network.Provider.Tungsten);
+                _networkServiceMapDao.persist(publicNetworkServiceMapVO);
+
+                Network managementNetwork = _networkModel.getSystemNetworkByZoneAndTrafficType(zoneId,
+                    Networks.TrafficType.Management);
+                NetworkServiceMapVO managementNetworkServiceMapVO = new NetworkServiceMapVO(managementNetwork.getId(),
+                    Network.Service.Connectivity, Network.Provider.Tungsten);
+                _networkServiceMapDao.persist(managementNetworkServiceMapVO);
+            }
+        });
+
+        SuccessResponse response = new SuccessResponse(getCommandName());
+        response.setDisplayText("config tungsten service successfully");
```

Comment:
spelling?  "configured tungsten service successfully".  not a very important change, but figured I would mention it as I noticed it.

Reply:
```suggestion
        response.setDisplayText("configered Tungsten-Fabric service successfully");
```